### PR TITLE
Fix violation of strict aliasing rule in `BaseImporter::CheckMagicToken`.

### PR DESCRIPTION
**Pull request by feuerste**
Previously when running a (failing) test locally without optimization and asan, I got a `sigill` with a stack trace to dereferencing the associated pointer in `BaseImporter::CheckMagicToken`. This is undefined behavior, see also the `strict aliasing` section in https://en.cppreference.com/w/cpp/language/reinterpret_cast#Notes, and can be fixed by `memcpy`ing instead of dereferencing.

### CODE REVIEW

**Comment by feuerste** [comment]
@kimkulling Would you mind to have a look at this one as well? This bytes us in our downstream tests...

**Comment by feuerste** [comment]
This is also related to #2147.

**Comment by feuerste** [comment]
@turol @kimkulling May I ask one of you for your review/merge of this PR? This fix is required for us downstream to upgrade to the latest master, as some of our tests are `sigill`ing without the fix... Thank you so much!

File: code/Common/BaseImporter.cpp [hunk 1]
```diff
@@ -339,15 +334,15 @@ std::string BaseImporter::GetExtension(const std::string &pFile) {
             // that's just for convenience, the chance that we cause conflicts
             // is quite low and it can save some lines and prevent nasty bugs
             if (2 == size) {
-                uint16_t rev = *magic_u16;
-                ByteSwap::Swap(&rev);
-                if (data_u16[0] == *magic_u16 || data_u16[0] == rev) {
+                uint16_t magic_u16;
```
[the remark this answers is not in the document]
Comment (kimkulling) [reply code/Common/BaseImporter.cpp:337]: Just checked, is fine.

**Review by kimkulling — APPROVED** [review]
Looks fine.

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**